### DISCUSSION TITLE
Using custom ConfigMapping with kubernetes-config extension

### DISCUSSION THREAD

**Discussion by bulldog98** [quarkusio/quarkus#33435] [Q&A] [answered]
I have a custom `ConfigMapping` annotated interface which I use in a bean. How can I get these values from the kubernetes-config extension without getting a stacktrace like this on startup (in kubernetes):

```
io.smallrye.config.ConfigValidationException: Configuration validation failed:
 java.util.NoSuchElementException: SRCFG00014: The config property custom.config is required but it could not be found in any config source
```

Note that in the configured configmap there is a value for `custom.config` and if I inject the value via `@ConfigProperty` it works

[a comment by an automated account is not shown]

**Reply by Sgitario** [reply]
If you get this failure, then the binding of the configMap has not been done. We would need a sample project / reproducer to assess what's going on.

**Reply by bulldog98** [reply]
@Sgitario I'll try to do a reproducer

**Reply by bulldog98** [reply]
@Sgitario I have a reproducer, but I'm not sure if that is minimal
[reproducer.zip](https://github.com/quarkusio/quarkus/files/11498688/reproducer.zip). The problem seems to be the default value

**Reply by bulldog98** [reply]
Definitally the default value in the config that is the problem, after setting that in the configmap it works in my real app

**Reply by geoand** [reply]
If you enable debug logging, do you see any ConfigMap related logs?

**Reply by bulldog98** [reply]
No I do not get any logging except the crash, my guess is that the `quarkus.log.level=DEBUG` is not applied since it's also a configuration @geoand

**Reply by geoand** [reply]
You can pass it as a system property

**Reply by bulldog98** [reply]
you mean with passing `-Dquarkus.log.level=DEBUG` to the binary? That also does not seem to work

**Reply by geoand** [reply]
Yes, that's what I mean

**Reply by bulldog98** [reply]
Yeah that does not work too

**Reply by bulldog98** [reply, marked as answer]
I found out what the issue was. I also had a custom `ConfigSourceInterceptor` that redirected `custom.config` to `custom-config` and then did not find the default value. I'm not sure if that is intended or if that is a bug.
